Library/Homebrew/test/spec_helper.rb

```diff
@@ -85,6 +85,10 @@
     skip "Not on macOS." unless OS.mac?
   end
 
+  config.before(:each, :needs_java) do
+    skip "Java not installed." unless which("java")
```

Comment:
This will behave oddly on macOS with the `java` shim but Java not being installed. Might want to use similar logic to the `JavaRequirement`.

Reply:
Good catch. Fixed.